third_party/move/move-core/types/src/account_address.rs

```diff
@@ -47,6 +47,46 @@ impl AccountAddress {
         Self(buf)
     }
 
+    /// Represent an account address in a way that is compliant with the v1 address
+    /// standard. The standard is defined as part of AIP-40, read more here:
+    /// https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-40.md
+    ///
+    /// In short, all special addresses MUST be represented in SHORT form, e.g.
+    ///
+    /// 0x1
+    ///
+    /// All other addresses MUST be represented in LONG form, e.g.
+    ///
+    /// 0x002098630cfad4734812fa37dc18d9b8d59242feabe49259e26318d468a99584
+    ///
+    /// For an explanation of what defines a "special" address, see `is_special`.
+    ///
+    /// All string representations of addresses MUST be prefixed with 0x.
+    pub fn to_standard_string(&self) -> String {
+        let suffix = if self.is_special() {
+            self.short_str_lossless()
+        } else {
+            self.to_canonical_string()
+        };
+        format!("0x{}", suffix)
+    }
```

Comment:
How will this interact with things like module addresses?

Currently, the compiler and other places will use the short string mode.  Some pieces are treated like strings and some are treated like addresses.

Reply:
Not 100% sure what you mean by this bit:

> Some pieces are treated like strings and some are treated like addresses.

Could you elaborate?

Reply:
Function IDs, struct names, all are converted to strings, and used as a string output / input.

This is the full `0x00000...01::coin::Coin` -> `0x1::coin::Coin`

This applies everywhere, so if that doesn't match it could be particularly problematic

Similarly a ton of other places in the compiler sometimes output long, sometimes short

Reply:
Right indeed. So this PR is not breaking in any way, it just adds a function that complies with AIP 40. Everywhere that accepts addresses should be updated to use the new from_str equivalent in their language in the next PR before we consider using this widely. Though there are places where I know we can start using this today without problems (new stuff, some particular configs / CLI stuff, etc) which is why I want to get it in.